docs/overview/changelog.md

```diff
@@ -1,9 +1,21 @@
 # Version history
 
-This page contains the release history for the Zulip Server 9.x stable
-release series. See the [current Zulip changelog][latest-changelog]
-for newer release series, or the [commit log][commit-log] for an
-up-to-date list of all changes.
+This page contains the release history for the Zulip server. See also the
+[Zulip release lifecycle](../overview/release-lifecycle.md).
+
+## Zulip Server 10.x series
+
+This section is an incomplete draft of the release notes for the next
+major release, and is only updated occasionally. See the [commit
+log][commit-log] for an up-to-date list of all changes.
+
+### Zulip Server 10.0 -- unreleased
```

Comment:
Having this info in the title as well as in the release date format below seems helpful for not being confused looking at the sidebar index.